topics-tools/gen-topics-database/main.go

```diff
@@ -4242,20 +4248,34 @@ func getTerminationNodeData(data *DataResponse, parentTopicNumber int, parentFul
 	return info
 }
 
-func getNodeData(data *DataResponse, parentTopicNumber int, parentFullURI string, bodyTextFile io.Writer, checkFile io.Writer) {
+func getNodeData(data *DataResponse, parentTopicNumber int, parentFullURI string, bodyTextFile io.Writer, checkFile io.Writer) bool {
+	var anyValid bool
```

Comment:
Consider explicitly setting this to false for clarity
```
anyValid := false
```
It makes it immediately clear to anyone reading this what the default/initial value of this var is. The zero value of a bool in Go is `false` but I wasn't 100% sure and had to look it up to make sure my understanding of this code was correct.

Reply:
I can see your point, but others would then pick me up saying to do it the way it currently is, so this one will be left as is.

Reply:
I would agree with dave, setting flags to their default state explicitly is much clearer